mlflow/genai/scorers/scorer_utils.py

```diff
@@ -82,59 +84,65 @@ def recreate_function(source: str, signature: str, func_name: str) -> Callable[.
         func_name: The name of the function.
 
     Returns:
-        The recreated function or None if recreation failed.
+        The recreated function.
+
+    Raises:
+        MlflowException: If function recreation fails (e.g., syntax errors, invalid signature).
     """
+    import mlflow
+
+    # Parse the signature to build the function definition
+    sig_match = re.match(r"\((.*?)\)", signature)
+    if not sig_match:
+        raise MlflowException(
+            f"Invalid signature format: '{signature}'", error_code=INVALID_PARAMETER_VALUE
+        )
+
+    params_str = sig_match.group(1).strip()
+
+    # Build the function definition
+    func_def = f"def {func_name}({params_str}):\n"
+    # Indent the source code
+    indented_source = "\n".join(f"    {line}" for line in source.split("\n"))
+    func_def += indented_source
+
+    # Create a namespace with common MLflow imports that scorer functions might use
+    # Include mlflow module so type hints like "mlflow.entities.Trace" can be resolved
+    import_namespace = {
+        "mlflow": mlflow,
+    }
+
+    # Import commonly used MLflow classes
     try:
-        # Parse the signature to build the function definition
```

Comment:
Just remove try / catch block and unindenting so that the exception is made very clear / visible to users who register scorers to remove servers